BUILD YOUR OWN SELLSWORD

Limeon666 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About the Pull Request
**I am very receptive to feedback. If you leave a comment I can change something.**

Lets you build your own sellsword from scratch. You can pick your armor class and weapon, and specialize in what you want. More information below.

Light Armor:
Same armor as base sellsword, but now gets dodge expert. This is also the only armor class that lets you pick between the typical backup dagger, or a lesser healing potion, or smoke bomb.

Medium Armor:
Hauberk, gambeson, sallet, gorget, tights, and the rest being standard sellsword gear. They have the best armor coverage to start with, and the medium armor trait.

Heavy Armor:
Light gambeson, chain chausses, coat of plates, dark boots, black leather gloves. This option has the heavy armor trait, but bad coverage. Their neck and head are exposed, and if their coat of plates breaks they are completely vulnerable. This option however has the most potential.

Weapon choices:
"Bastard Sword", "Billhook", "Dual Daggers", "Flail", "Steel Axe", "Iron Mace", "Rapier", "Saber", "Short Sword", "Spear", "Cutlass", "Crossbow", "Bow".

Your weapon choice affects what weapon skills you have, but no choice gives you more skill points overall than prior. The crossbow and bow choices also give you a loaded quiver, and one less strength with the gain of one perception, which is important for ranged weapons.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

This change lets Sellsword represent the generic mercenary, and lets people customize it to their heart's delight. It also gives them a small boost, which is justified given how weak they are compared to any other mercenary. More customization and choices in class selection is good, and the amount of customization gives them a solid reason for existing while still keeping them fairly weak.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->


## Proof of Testing (Required)
### LIGHT + DAGGERS 
![lightdagger3](https://github.com/user-attachments/assets/33a00d23-8973-4fd0-9b67-63d5a03a39c6)
### MEDIUM + CROSSBOW
![mediumcrossbow2](https://github.com/user-attachments/assets/1b0f3326-636e-46df-89df-4a9f1480bedf)
### HEAVY + STEEL AXE
![heavyaxe2](https://github.com/user-attachments/assets/a76fdc92-91e9-4adf-ae77-d3a5bf7fd0ff)

Closed this on accident lol. Also ignore the amount of commits. I goofed.
<!-- Show proof of testing, screenshots or recordings for features, proof of compile for backend changes -->

Limeon666 commented:
Apologies for the amount of commits, by the way, I gooft

Genevesh commented:
Looks amazing, let's hope nobody makes invincible murderhobo minmaxxers

john-pawson commented:
While the light armor sellsword is fine, and could honestly use a little better armor, the medium and heavy ones get *way* more armor than they should for being mercenaries on the first place, the medium armor shouldn't include chain leggings on top of the gambeson and hauberk and, the heavy armor probably shouldn't include plated leggings *with* both haubergeon and half-plate, compared to all of the other mercenaries, in which they usually have at least 2-3 important bodyparts exposed.

The grenzel gets an exposed head and legs, desert gets exposed head, arms and neck, newmoon get exposed legs, neck and arms (low quality head prot), and forlorn gets exposed arms and legs.

While on this one, medium only gets exposed face, which would be fine considering that the armors like gambesons and hauberks break after a bit of fighting, if they didn't have a third layer on the legs that's the chain chausses, meaning that hits to the leg no longer contribute to the overall armor damaging.
And heavy only gets exposed neck and head, while having the best possible armor for chest and legs, being quite literally a direct upgrade from desert riders, just to give an example.

Other than that balancing conundrum, great PR.

notafurryy commented:
Battle axe shouldnt be a roundstart weapon, it can delimb through chainmaille with one hit.

Also knives 4 is too much for a merc, if you pick a fast race, elves, kobolds, tabaxi, verminfolk you reduce your opponents parry chance to zero, there is a reason not even rogues get it and they get a str debuff even with knives 3

Limeon666 commented:
> Also knives 4 is too much for a merc, if you pick a fast race, elves, kobolds, tabaxi, verminfolk you reduce your opponents parry chance to zero, there is a reason not even rogues get it and they get a str debuff even with knives 3

Ok, would changing it to knives 3, and regular steel axe work?

DelineFortune commented:
Axe 4 is pretty op, no role has is as far I know
With a battle steel axe (70 ap), it allows you to cut knights through armor against av player

Limeon666 commented:
Nerfed knife skill to 3 for paired daggers, replaced the war axe with a steel axe, nerfed medium/heavy armor a bit, and updated the PR page to be up to date. Thanks for the feedback everyone.

Sputes commented:
This is awesome! Only feedback I've got is the one handed weapons ought to get a wooden shield since they're a bit hard to craft.

Purpleturtle3 commented:
If this pr gets merged then desert riders will be weakest mercs by far

john-pawson commented:
> If this pr gets merged then desert riders will be weakest mercs by far

TAdesert merc buff might be a good idea for later.
But.. they *do* get 2 SPD, 2 STR and 2 END, which are probably the three most important stats for combat.

notafurryy commented:
> If this pr gets merged then desert riders will be weakest mercs by far
> 

They are the only merc to have the torso and all limbs covered, also their sword has long reach so they can stab feet.

Purpleturtle3 commented:
> > If this pr gets merged then desert riders will be weakest mercs by far
> 
> They are the only merc to have the torso and all limbs covered, also their sword has long reach so they can stab feet.

"Torso and all limbs covered"
![image](https://github.com/user-attachments/assets/ae406f6f-33d8-4367-834e-0831c46e3a96)
There is no arm protection here, nor neck or head.
![image](https://github.com/user-attachments/assets/f2cb8c4d-7da4-45af-8548-89fb5dffcde0)
Meanwhile heavy armor merc starts with coat of plates CHEST|GROIN|VITALS|ARMS, STEEL chain chausses, which mind you, desert rider has IRON ones.
![image](https://github.com/user-attachments/assets/76e17c5b-2203-4546-960d-c18efc264384)

As for the desert rider sword? It's just a barstard sword with a bit more damage and it can't stab, these are the weapons that the heavy merc can pick
![image](https://github.com/user-attachments/assets/d3ced3db-b462-4c46-9c55-018f4b5ae93b)
Notice something? Two weapons that can hit feet, with the possibility to get weapon skill in weapons that are far more dangerous than swords, like... axes, polearms, maces...

Yes, a desert rider has more two stats, but they can't get heavy armor, can't use maces, spears, etc.

Nevermind you can also pull up a dodge expert light armor merc with a rapier roundstart.

Limeon666 commented:
@Purpleturtle3 
While I think you may be right that if this change went through desert riders would be slightly weak comparatively, do keep in mind sellswords are currently the weakest mercs by far, and desert riders do still at least hold stat advantages which is pretty relevant.

With that in mind what are your specific suggestions? I can potentially tweak things.

notafurryy commented:
> Meanwhile heavy armor merc starts with coat of plates CHEST|GROIN|VITALS|ARMS, STEEL chain chausses, which mind you, desert rider has IRON ones.

Huh, thought that scalemail protected the arms too, also isn't the whole thing with the heavy merc archetype is it's lack of coverage? odd. Though it's more of the case of the heavy sellsword needing a nerf than the other way around.

Limeon666 commented:
> > Meanwhile heavy armor merc starts with coat of plates CHEST|GROIN|VITALS|ARMS, STEEL chain chausses, which mind you, desert rider has IRON ones.
> 
> Huh, thought that scalemail protected the arms too, also isn't the whole thing with the heavy merc archetype is it's lack of coverage? odd. Though it's more of the case of the heavy sellsword needing a nerf than the other way around.

The whole thing with heavy sellsword is if their coat of plates, which only has 250 durability, breaks they're fucked. That being said I can probably replace the chain pants to iron ones, since giving them steel wasn't intentional, but steel ones only have 25 more durability. That's the only difference I can gleam. Heavy sellsword also has no neck, head, face protection.

notafurryy commented:
> That being said I can probably replace the chain pants to iron ones, since giving them steel wasn't intentional

Ideally they would only have chest protection, maybe also a mask.

> Heavy sellsword also has no neck, head, face protection

That's also the case with the desert rider.

Limeon666 commented:
> > That being said I can probably replace the chain pants to iron ones, since giving them steel wasn't intentional
> 
> Ideally they would only have chest protection, maybe also a mask.
> 
> > Heavy sellsword also has no neck, head, face protection
> 
> That's also the case with the desert rider.

Yes, and again, Desert Riders get double the bonus in speed and endurance stats, and while shalal is bad headgear/neckgear it's still armor there. I think it's fine to give heavy very mediocore heavy armor, for the cost of that.

Iron Chainmail is the absolute worst leg armor I could give heavy, for it to still make sense. Leather trousers + plate just doesn't work aesthetically or thematically, and it really isn't too strong for them to get basic chain leggings.

Vasili59 commented:
Will this PR be accepted?

Sirgir commented:
Would it be too much to ask to add whips to the list of weapon choices? Unconventional mercenaries are my jam-- I particularly like the Zybantine Whip Mercenaries... so it would be fun to be a whip merc without having to be a slaver.

JohnnyJohnJohnson commented:
PLEASE accept and merge this, this idea is awesome.

RadiantFlash7 commented:
I'll review this PR relatively soon. In the meantime fix the merge conflicts you can @Limeon666

Sirgir commented:
Hey if you're still open to taking weapon suggestions, why not add a bardiche on there as well? It's rarely ever seen...

Vasili59 commented:
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO!  
NOT THAT!
MY FAVORITE PR!
AUTHOR, WAKE UP, PLEASE!

Genevesh commented:
Someone must urgently take over this PR

JohnnyJohnJohnson commented:
For fuck's sake

Gyrandola commented:
Closing this. See linked PR for reasons.
